src/backend/distributed/commands/schema.c

```diff
@@ -74,6 +76,19 @@ PreprocessCreateSchemaStmt(Node *node, const char *queryString,
 
 	commands = list_concat(commands, GetGrantCommandsFromCreateSchemaStmt(node));
 
+	if (ShouldUseSchemaAsTenantSchema(createSchemaStmt->schemaname))
```

Comment:
name is a bit hard to follow, maybe ShouldUseSchemaSharding / ShouldUseSchemaBasedSharding?